### PR TITLE
Expand test coverage of PaymentInformation action creators

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Greatly improved the test coverage of our PaymentInfo action creators. Highlights:

- make sure we're calling the correct endpoints with the correct methods
- test both the happy and sad paths
- check that we're reporting data to Google Analytics

## Testing done
🤔 

## Screenshots
N/A

## Acceptance criteria
- [ ] Uhhh... better tests

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile360/actions/paymentInformation.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const PAYMENT_INFORMATION_SAVE_SUCCEEDED =
 export const PAYMENT_INFORMATION_SAVE_FAILED =
   'PAYMENT_INFORMATION_SAVE_FAILED';
 
-export function fetchPaymentInformation() {
+export function fetchPaymentInformation(recordEvent = recordAnalyticsEvent) {
```
Comment (erikphansen) [inline src/applications/personalization/profile360/actions/paymentInformation.js]: Using dependency injection here so I test that the `recordEvent` helper is called correctly. I'm open to other ideas to make the calls to `recordEvent` testable.

File: src/applications/personalization/profile360/tests/actions/paymentInformation.unit.spec.js [hunk 2]
```diff
@@ -1,27 +1,25 @@
...
-      ok: true,
-      json: () =>
-        Promise.resolve({
-          data: {
-            attributes: {},
-          },
-        }),
-    }),
-  );
+  mockFetch();
```
Comment (erikphansen) [inline src/applications/personalization/profile360/tests/actions/paymentInformation.unit.spec.js]: switched to using our standard `mockFetch` test helpers.

File: src/applications/personalization/profile360/actions/paymentInformation.js (hunk 1 above)
Comment (ncksllvn) [reply src/applications/personalization/profile360/actions/paymentInformation.js]: I sometimes add an underscore to the import to help draw the distinction -

```
export function fetchPaymentInformation(recordAnalyticsEvent) = _recordAnalyticsEvent) {
```

**Review by ncksllvn — APPROVED** [review]

File: src/applications/personalization/profile360/actions/paymentInformation.js (hunk 1 above)
Comment (erikphansen) [reply src/applications/personalization/profile360/actions/paymentInformation.js]: Interesting idea

**Review by rianfowler — APPROVED** [review]

**Pull request merged**